src/main/java/com/exactpro/epfast/decoder/integer/DecodeMandatoryInt32.java

```diff
@@ -9,24 +9,53 @@
 
     private static final int NEGATIVE_LIMIT = Integer.MIN_VALUE >> 7;
 
+    private boolean checkForSignExtension = false;
+
     private int value;
 
     public void decode(ByteBuf buf) {
         reset();
+        checkForSignExtension = false;
         int readerIndex = buf.readerIndex();
         int readLimit = buf.writerIndex();
         int oneByte = buf.getByte(readerIndex++);
         if ((oneByte & SIGN_BIT_MASK) == 0) {
             value = 0;
             accumulatePositive(oneByte);
-            while (readerIndex < readLimit && !ready) {
-                accumulatePositive(buf.getByte(readerIndex++));
+            if (oneByte < 0) {
+                buf.readerIndex(readerIndex);
+                return;
+            }
+            if (buf.readableBytes() > 1) {
```

Comment:
We already have reader and writer indexes. Readable bytes is writerIndex - readerIndex. Plus you need just one byte here. So it should be `> 0`